src/drawing.rs

```diff
@@ -360,4 +361,72 @@ impl<'a> Pencil<'a> {
         }
         self.move_origin(-position)
     }
+
+    /// Draws one of the frames of the given `animator` based on the number of times this has
+    /// been previously called.
+    pub fn draw_animator(&mut self, animator: &mut Animator, position: Vec2) {
+        self.draw_text(&animator.access_frame(), position);
+    }
+}
+
+/// An object that runs an [`Animation`].
+pub struct Animator {
+    animation: Vec<AnimationFrame>,
+    frame_index: usize,
+    counter: i32,
+    speed: u32,
+}
+
+impl Animator {
+    /// Creates a new [`Animator`] with the given `frames` and with `rate` number of frames
+    /// displayed per animation frame.
+    pub fn new(animation: Vec<AnimationFrame>) -> Animator {
+        Animator {
+            animation,
+            frame_index: 0,
+            counter: 0,
+            speed: 1,
+        }
+    }
+
+    /// Sets the speed of animation.
+    pub fn set_speed(&mut self, speed: u32) {
+        self.speed = speed
+    }
+
+    /// Gets the text of the current frame and updates the frame counter, changing the current
+    /// frame if the duration is exceeded.
+    fn access_frame(&mut self) -> String {
+        let current_frame = &self.animation[self.frame_index];
+
+        self.counter += 1;
+        if self.counter >= current_frame.duration / self.speed as i32 {
+            self.counter = 0;
+            self.frame_index = (self.frame_index + 1) % self.animation.len();
+        }
+        if self.frame_index >= self.animation.len() {
+            self.frame_index = 0;
+        }
+
+        current_frame.text.clone()
+    }
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct AnimationFrame {
+    pub(crate) text: String,
+    duration: i32,
+}
+
+impl AnimationFrame {
+    pub fn new(text: &str, duration: i32) -> AnimationFrame {
+        let text = text.to_string();
+        AnimationFrame { text, duration }
+    }
+}
+
+impl Display for AnimationFrame {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.text)
+    }
 }
```

Comment:
Is this implementation really needed? I mean, does it makes sense to display an `AnimationFrame`?